packages/varnish/src/components/social/ShareOnTwitterButton.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import IconButton from '@mui/material/IconButton';
+import TwitterIcon from '@mui/icons-material/Twitter';
+
+interface Props {
+  url?: string;
+  text?: string;
+}
+
+export function ShareOnTwitterButton({ url, text }: Props) {
+  let baseUrl = 'https://twitter.com/intent/tweet';
+  if (url) {
+    baseUrl += `?url=${  url}`;
+    if (text) {
+      baseUrl += `&text=${  text}`;
+    }
+  } else if (text) {
+    baseUrl += `?text=${  text}`;
+  }
+
+  return (
+    <IconButton color="primary" size="small">
```

Comment:
we might want to change the color on hover...
see the top of: https://delphi.allenai.org/
(note, we do it differently elsewhere.. so we should ask @aaronsarnat)
here is no hover (bottom):
https://satlas-explorer.apps.allenai.org/

Reply:
let me get this in and consult with Aaron regarding any changes! if this gets merged then maybe he can play with it in its current form and suggest ideas based on that